packages/headless/src/app/engine-configuration.ts

```diff
@@ -155,7 +155,17 @@ export interface AnalyticsConfiguration {
    * @internal
    */
   trackingId?: string;
+  /**
+   * Specifies the analytics mode to use.
+   * By default, `legacy`.
+   * @internal
+   */
   analyticsMode?: 'legacy' | 'next';
+  /**
+   * The version of `@coveo/atomic` used.
+   * @internal
+   */
+  atomicVersion?: string;
```

Comment:
This feels like cheating, by adding a key that's named specifically "for us"...

What about Quantic, or a partner building/investing into a UI layer with Headless ?

How about naming it generically ? 
```js
uiVersion: {name: string; version: string; owner: string; }

// eg:
// uiVersion: {name: 'atomic', version: '1.2345', owner: '@coveo'}
// uiVersion:{name: 'foo', version: '4.5678', owner: '@acme.inc'}
```

And then we can adapt `getAnalyticsSource` to take that generic object and format it properly

Reply:
So the 'for us' here is by design, but I scoped it too narrowly, you're right.
By that, I mean we want to allow reporting only on our framework, to ensure a certain purity of the `source` field for the analytics.
However, you're right that it should not be Atomic-Exclusive.
There'll be an allow-list of frameworks (for now, Quantic & Atomic).

However, I'll sync up with DP soon because it's been 2 months ish since I had those discussion and my memory might be a bit fuzzy

Reply:
So, thought about it while doing the change in the implementation and verifying the background work done in the backend.
I see nothing blocking preventing people from indicating their custom framework in the source parameter.
However, I do prefer to take the cautious route and keep the API narrow and extend it if the need arises, a
So, I'd keep things as they currently are as of 3a646c9 for the time being

If we need to add singular keys, we can adopt a punch-hole strategy and allow them one by one.
If we need to open the gate and allow more, the format is extensible and should allow us to allow custom frameworks to hook themselves. 

(moreover, going from loose to strict is hard to do without breaking changes/degradation of Headless API)